app/controllers/restful_redirect_controller.rb

```diff
@@ -13,7 +13,12 @@ def index
         elsif %w[ManageIQ::Providers::EmbeddedAnsible::AutomationManager].include?(record.type)
           redirect_to(:controller => 'ansible_playbook', :action => 'show_list')
         else
-          redirect_to(polymorphic_path(record))
+          begin
+            redirect_to(polymorphic_path(record))
+          rescue NoMethodError
+            flash_to_session(_("Cannot redirect to \"%{record}\" provider.") % {:record => record.name}, :error)
+            redirect_to(:controller => 'ops', :action => 'explorer')
```

Comment:
I don't think we need to redirect user back to the tenant detail, but can stay in the screen displaying providers

Reply:
After discussion with Martin P we decided to redirect user back to the Tenant textual summary.